ginga/util/iqcalc_astropy.py

```diff
@@ -0,0 +1,321 @@
+"""Module to handle image quality calculations using ``astropy``."""
+
+import numpy as np
+from astropy.modeling import models, fitting
+
+from ginga.misc import Bunch
+
+# Reuse shared items from the old module until we can merge old and new together.
+from ginga.util.iqcalc import IQCalcError, get_median, have_scipy
+from ginga.util.iqcalc import IQCalc as _IQCalc
+
+# Import the rest into namespace so we can use this module like iqcalc.
+from ginga.util.iqcalc import get_mean  # noqa
+
+__all__ = ['IQCalc']
+
+
+# TODO: Use photutils for source finding.
+class IQCalc(_IQCalc):
+    """This is `ginga.util.iqcalc.IQCalc` that uses ``astropy``.
+
+    This subclass has an extra ``self.fitter`` attribute for ``astropy``
+    fitting.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fitter = fitting.LevMarLSQFitter()
+
+    # FWHM CALCULATION
+
+    def _prep_for_fitting(self, arr1d, medv):
+        if not have_scipy:
+            raise IQCalcError("Please install the 'scipy' module "
+                              "to use this function")
+
+        N = len(arr1d)
+        X = np.array(list(range(N)))
+        Y = np.asarray(arr1d)
+
+        # Fitting works more reliably if we do the following
+        # a. subtract sky background
+        if medv is None:
+            medv = get_median(Y)
+        Y -= medv
```

Comment:
If I change this to:
```python
Y = Y - medv
```
then it solves the problem of side-effecting the cutout array.

Reply:
Oh, weird... Maybe its `isub` isn't behaving as expected.

Reply:
I suspect that it depends on what kind of array you are using it with. If the cutout ends up just being a view into the cutout, then doing `-=` probably drops the hatchet on the cutout.

Reply:
Good to know! Sorry about that.